contracts/keepers/FliRebalanceKeeper.sol

```diff
@@ -0,0 +1,79 @@
+/*
+    Copyright 2021 Index Cooperative.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { KeeperCompatibleInterface } from "@chainlink/contracts/src/v0.6/KeeperCompatible.sol";
+
+/**
+ * @title RebalanceKeeper
+ * @author Index Cooperative
+ * 
+ * Chainlink Keeper which automatically rebalances FLI SetTokens.
+ */
+contract FliRebalanceKeeper is KeeperCompatibleInterface {
+    using Address for address;
+
+    /* ============ Modifiers ============ */
+    modifier onlyRegistry() {
+        require(msg.sender == registryAddress, "Only registry address can call this function");
+        _;
+    }
+
+    /* ============ State Variables ============ */
+
+    address public fliExtension;                          // Address of the fli extension contract
+    address public registryAddress;                       // Address of the chainlink keeper registry
+
+    /* ============ Constructor ============ */
+    constructor(address _fliExtension, address _registryAddress) public {
+        fliExtension = _fliExtension;
+        registryAddress = _registryAddress;
+    }    
+
+    /**
+     * As checkUpkeep is not a view function, calling this function will actually consume gas.
+     * As such if a keeper calls this function, it will always return true so that performUpkeep will be called.
+     */    
+    function checkUpkeep(bytes calldata /* checkData */) external override onlyRegistry returns (bool, bytes memory) {
```

Comment:
Even though checkUpkeep isn't a view function, it is called as if it were by the keepers. This is because you can call a non-view method and it runs as if it were executed, except state changes are not reflected. We can then just have this function do the same thing as `getRebalanceCalldata`. This will save us on gas since it can be run off-chain.

Finally I don't think we need to make this onlyRegistry since it won't perform any gatekeeped actions.

Reply:
done - removed it